### PR TITLE
feat(bulk-import): upgrading to mui5

### PR DESCRIPTION
**Pull request by its-mitesh-kumar**
Resolves : https://issues.redhat.com/browse/RHIDP-4308

### CODE REVIEW

**Comment by its-mitesh-kumar** [comment]
cc :  @debsmita1

**Comment by 04kash** [comment]
@its-mitesh-kumar To fix CI issues, you'd have to run `yarn dedupe` from the bulk-import workspace

File: workspaces/bulk-import/plugins/bulk-import/src/components/AddRepositories/AddRepositories.tsx
Comment (debsmita1) [inline workspaces/bulk-import/plugins/bulk-import/src/components/AddRepositories/AddRepositories.tsx:53]: Can you comment this for now instead of removing. We would need this once the `ServiceNow` approval tool is incorporated in the plugin

**Comment by debsmita1** [comment]
Tested the changes locally, and the changes look fine
@its-mitesh-kumar Please fix the Resize observer issue with the side panel

**Comment by its-mitesh-kumar** [comment]
@divyanshiGupta 
Resize observer issue @debsmita1  has mentioned a warning that will be shown only in dev mode . 
Screenshot 2024-11-13 at 3 41 39 PM
Cc : @christoph-jerolimov

**Comment by divyanshiGupta** [comment]
If we aren't using ResizeObserver in the code we can maybe ignore this issue for this PR as it will not appear in prod but it will be still better to fix it in the dev mode in this PR if possible and if not then we can merge this PR and raise another for the dev mode once we have a solution.

**Comment by its-mitesh-kumar** [comment]
@divyanshiGupta ! With the below commit resize issue has gone .
https://github.com/redhat-developer/rhdh-plugins/pull/24/commits/69036a1f8d9dc64df2f3bf9cadf315f2cf839290

File: workspaces/bulk-import/packages/app/src/components/search/SearchPage.tsx
Comment (divyanshiGupta) [inline workspaces/bulk-import/packages/app/src/components/search/SearchPage.tsx:48]: Remove or uncomment

File: workspaces/bulk-import/plugins/bulk-import/src/components/AddRepositories/AddRepositories.tsx
Comment (divyanshiGupta) [reply workspaces/bulk-import/plugins/bulk-import/src/components/AddRepositories/AddRepositories.tsx:53]: @its-mitesh-kumar please add comment describing why this code is commented out.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Comment by christoph-jerolimov** [comment]
I still have the resize issue; it depends on how your OS renders the scrollbar.

The issue is logged here in MUI: https://github.com/mui/material-ui/issues/43718 (there are also multiple duplicates)

For me, this issue appears only when multiple TextFields with the `multiline` prop (an auto-resizing text field) are used at the same time. But that's exactly what we want here.

I think we can go ahead for now.

**Review by divyanshiGupta — APPROVED** [review]
/lgtm

**Comment by christoph-jerolimov** [comment]
I noticed one difference, can we check if we can fix this?

The background color in dark mode changed:

Before:

before

With this PR:

after

**Comment by its-mitesh-kumar** [comment]
@christoph-jerolimov ! For me its looking like below . 
Screenshot 2024-11-14 at 4 11 51 PM

**Comment by christoph-jerolimov** [comment]
Don't understand how this is possible, but we can start merging and then we will take a look again.

**Review by christoph-jerolimov — APPROVED** [review]

**Pull request merged**
